### PR TITLE
Feature/mechanist

### PR DESCRIPTION
**Pull request by Raycoms**
# Changes proposed in this pull request:
- Adds mechanic worker
- Improves some handling of requests of smelters
- Only allow overruling of requests that are assigned of the player (Rest is done via GUI still)

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/mechanic/EntityAIWorkMechanic.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ * Crafts everything else basically (redstone stuff etc)
+ */
+public class EntityAIWorkMechanic extends AbstractEntityAICrafting<JobMechanic>
+{
+    /**
+     * Initialize the mechanic and add all his tasks.
+     *
+     * @param sawmill the job he has.
+     */
+    public EntityAIWorkMechanic(@NotNull final JobMechanic sawmill)
```
Comment (marchermans) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/mechanic/EntityAIWorkMechanic.java]: Not a sawmill.

**Review by marchermans — CHANGES_REQUESTED** [review]
The RS Changes are okey, as per discussion.
However i would check your copy pasting work.....

**Comment by LDTTeam-Buildsystem** [comment]
This PR has been automatically built and hosted by our build system, during this process there was a crash, error, or otherwise, please check on this at: https://teamcity.minecolonies.com/viewLog.html?buildId=39595&buildTypeId=LetSDevTogether_Minecolonies_PullRequests_2_DeployTestServer

**Comment by LDTTeam-Buildsystem** [comment]
This PR has been automatically built and hosted by our build system, during this process there was a crash, error, or otherwise, please check on this at: https://teamcity.minecolonies.com/viewLog.html?buildId=39617&buildTypeId=LetSDevTogether_Minecolonies_PullRequests_2_DeployTestServer

**Review by someaddons — APPROVED** [review]

**Pull request merged**